src/Domain/Shared/ValueObject/AbstractUuid.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Shared\ValueObject;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class AbstractUuid
+{
+    /**
+     * @return static
+     */
+    public static function fromUuid(UuidInterface $value): self
+    {
+        return new static($value);
+    }
+
+    public function toUuid(): UuidInterface
+    {
+        return $this->value;
+    }
+
+    /**
+     * @return static
+     */
+    public static function fromBytes($bytes): self
+    {
+        return new static(Uuid::fromBytes($bytes));
+    }
+
+    /**
+     * @return static
+     */
+    public static function fromString($name): self
+    {
+        return new static(Uuid::fromString($name));
+    }
+
+    /**
+     * @return static
+     */
+    public static function fromInteger($integer): self
+    {
+        return new static(Uuid::fromInteger($integer));
+    }
+
+    /**
+     * @param null|mixed $node
+     * @param null|mixed $clockSeq
+     *
+     * @throws \Exception
+     *
+     * @return static
+     */
+    public static function uuid1($node = null, $clockSeq = null): self
+    {
+        return new static(Uuid::uuid1($node, $clockSeq));
+    }
+
+    /**
+     * @return static
+     */
+    public static function uuid3($ns, $name): self
+    {
+        return new static(Uuid::uuid3($ns, $name));
+    }
+
+    /**
+     * @throws \Exception
+     *
+     * @return static
+     */
+    public static function uuid4(): self
+    {
+        return new static(Uuid::uuid4());
+    }
+
+    /**
+     * @return static
```

Comment:
please remove all this redundant comments

Reply:
removed but then returned annotations because phpstan think that code return AbstractString|AbstractUuid instead inheritors